src/users.cr

```diff
@@ -75,4 +75,44 @@ module Office365::Users
   def list_users(response : HTTP::Client::Response)
     UserQuery.from_json response.body
   end
+
+  def create_user_from_json(string_or_io) : User
+    request = graph_http_request("POST", "/v1.0/users", data: string_or_io)
+    response = graph_request(request)
+
+    get_user(response)
+  end
+
+  def update_user_from_json(id, string_or_io)
+    request = graph_http_request("PATCH", "/v1.0/users/#{id}", data: string_or_io)
+    graph_request(request)
+  end
+
+  def delete_user(id : String)
+    request = graph_http_request("DELETE", "/v1.0/users/#{id}")
+    graph_request(request)
+  end
+
+  def list_users_by_query(query : Hash(String, String)? = nil, q : String? = nil)
+    # https://graph.microsoft.com/v1.0/f50e8d54-1202-4c05-a58a-4ab4331964d2/users?$filter=id in ('5e44b9ff-b1ff-4656-8b80-6a365ff3dce1', '255bcc9c-354f-4978-8b14-75ed3f6eeb06')
+
+    path = "/v1.0/users"
```

Comment:
Is this the same base path for all calls in this scope? Perhaps extract to a constant and template into other methods?

Reply:
good idea